atlassian_jwt_auth/signer.py

```diff
@@ -43,15 +43,15 @@ def _generate_claims(self, audience, **kwargs):
     def _now(self):
         return datetime.datetime.utcnow()
 
-    def generate_jwt(self, audience, **kwargs):
+    def generate_jwt(self, audience, return_claims=False, **kwargs):
         """ returns a new signed jwt for use. """
         key_identifier, private_key_pem = self.private_key_retriever.load(
             self.issuer)
-        return jwt.encode(
-            self._generate_claims(audience, **kwargs),
-            key=private_key_pem,
-            algorithm=self.algorithm,
-            headers={'kid': key_identifier.key_id})
+        claims = self._generate_claims(audience, **kwargs)
+        encoded_jwt = jwt.encode(dict(claims), key=private_key_pem,
+                                 algorithm=self.algorithm,
+                                 headers={'kid': key_identifier.key_id})
+        return (encoded_jwt, claims) if return_claims else encoded_jwt
```

Comment:
What's the performance penalty on decoding without verifying/validating the generated jwt ? (Also, I have a strong preference for making methods always return results in the same form - so I'd prefer to change this to always return a tuple of claims and the generated jwt).

Reply:
If always returning a tuple won't break code that uses this library then I am all for doing that.

I am not sure about the performance of decoding, but it doesn't sit right with me.